### DISCUSSION TITLE
Better Mutation Support?

### DISCUSSION THREAD

**Discussion by reilnuud** [sebastian-lenz/craft-linkfield#215] [General]
Not sure if you're taking feature requests but we've been using the GraphQL mutations on a lot of projects lately (especially when translating sites); we have a system for converting link objects in to JSON strings to mutate before pushing, but it's been a lot of guesswork to see how the content needs to be formatted and we haven't even tried tackling the entry-type links yet.

For example, the object retrieved from GraphQL may look like:

```javascript
{
  url: "mailto:email@sample.com"
  text: "email@sample.com"
  type: "email"
}
```

But the mutation requires a valid email as the link content so we need to push something like (as a JSON string):

```json
{ "type": "email", "linkedUrl": "email@sample.com", "customText": "email@sample.com"}
```

It would be much nicer if there was a 1:1 system for mutating so we can push objects up to the GraphQL mutation API with the same schema that we retrieve them.

**Comment by sebastian-lenz** [comment]
Which version of the plugin are you using as we've extended the output of the plugin quite a bit for version 2 which is currently a release candidate. Due to my time constraints I am not able to implement new features currently.

**Comment by ce2cs** [comment]
> Not sure if you're taking feature requests but we've been using the GraphQL mutations on a lot of projects lately (especially when translating sites); we have a system for converting link objects in to JSON strings to mutate before pushing, but it's been a lot of guesswork to see how the content needs to be formatted and we haven't even tried tackling the entry-type links yet.
> 
> For example, the object retrieved from GraphQL may look like:
> 
> ```js
> {
>   url: "mailto:email@sample.com"
>   text: "email@sample.com"
>   type: "email"
> }
> ```
> 
> But the mutation requires a valid email as the link content so we need to push something like (as a JSON string):
> 
> ```json
> { "type": "email", "linkedUrl": "email@sample.com", "customText": "email@sample.com"}
> ```
> 
> It would be much nicer if there was a 1:1 system for mutating so we can push objects up to the GraphQL mutation API with the same schema that we retrieve them.

Also using GraphQL mutation on typed link field. Which version are you using? I'm on 1.0.25, I only have a `cta` field in the mutation. I cannot change `customText` field. We are trying to upgrade to craft 4 and typed link field 2.x. May I know is the `customText` usable in mutation?
